site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,394 @@
+
+A `Service` may only be specified as a `parentRef` of an `HTTPRoute` if it is:
+
+* NOT already specified as a `parentRef` of another `HTTPRoute` [in the same namespace]
+    * Resolving conflicts from multiple `HTTPRoutes` attempting to target the same `Service` as a `parentRef` should follow the existing [conflict guidelines](https://gateway-api.sigs.k8s.io/concepts/guidelines/#conflicts).
+      * When an existing `HTTPRoute` targeting the `Service` as a `parentRef` has already been accepted, any `HTTPRoute` with a newer creation timestamp should be rejected, because the match target considered for the specificity rules (the `Service`) is identical.
+      * If creation timestamps are identical (from applying both routes simultaneously), only the `HTTPRoute` with a `name` appearing first in alphabetical order should be accepted.
+      * Any rejected `HTTPRoute` should set an `Accepted` condition with `status: "False"`.
+    * This requirement may be relaxed later to follow the existing [precedence rules for merging configuration](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule), but is intended to simplify initial implementation.
+    * The namespace colocation requirement effectively forecloses using this strategy to delegate route management, but there is service mesh [user interest](https://github.com/istio/istio/issues/22997) in merging separately-defined routes together to manage large configurations, multiple versions or canary deployments for a single mesh service.
+* NOT already specified as a `backendRef` of any _other_ `HTTPRoute` with a `Service` `parentRef`
+   * This restriction is intended to prevent route "inclusion/delegation" patterns or circular references from multiple tiers of routing rules _within mesh configuration_.
+   * A `Service` MAY still be listed as a `backendRef` on a different `HTTPRoute` which only targets a `Gateway` as a `parentRef`. The specific behavior of how Gateways might interact with mesh routing rules will be defined in a future GEP.
+    * A `Service` MAY still be listed as a `backendRef` on the _same_ `HTTPRoute` where it is specified as a `parentRef`, e.g. to only split off some subset of traffic from the default backend endpoints.
+    * This could be computationally expensive to validate.
+
+This should be enforced by controller validation, which should set an `Accepted: { status: False }` condition with a corresponding new `ParentRefConflict` `RouteConditionReason`.
+
+The logic described here is a bit verbose, but could be validated with conformance tests, and is intended to allow an `HTTPRoute` configuring mesh traffic rules to additionally specify a `Gateway` `parentRef` (or `HTTPRoute` `parentRef` if [GEP-1058: Route Inclusion and Delegation](https://github.com/kubernetes-sigs/gateway-api/pull/1085) moves forward) to apply the same routing rules when exposing a mesh service outside the cluster.
```

Comment:
I think it is a bit of overkill - a user may deploy an envoy/nginx/etc as a service an have it route to whatever they want, creating cycles anyways. 

I would add a section about 'mesh implementation should implement the Forwarded header, and support a max number of forwards to prevent cycles'. This also provides a standard way for apps to detect the path of the request.

Reply:
Hmmm, that is an interesting option - I was definitely leaning towards a UX that could validate this when config is created to set a status, but it's definitely more of a pain to implement.